functions/api/product.js

```diff
@@ -1,14 +1,14 @@
 const constants = require('../common/constants')
 const validate = require('../common/validator')
 const logger = require('../middleware/logger');
+const { isAdmin, isAuthenticated } = require('../middleware/auth')
```

Comment:
can we remove **isAuthenticated** everywhere else too, if its being internally used in isAdmin. One suggestion is to authenticate for every api and have isAdmin(authorization check ) for create ,update,delete. So in startup.js we can add **isAuthenticated as a middleware** so that it applies for every api and in **isAdmin we could remove the isAuthenticated check** as it already taken care.